tsl/test/sql/include/continuous_aggs_ddl_common.sql

```diff
@@ -1107,3 +1107,98 @@ AS
 
 ALTER MATERIALIZED VIEW i3696_cagg2 SET (timescaledb.materialized_only = 'true');
 
+--TEST test with multiple settings on continuous aggregates --
+-- test for materialized_only + compress combinations (real time aggs enabled initially)
+CREATE TABLE test_setting(time timestamptz not null, val numeric);
+\if :IS_DISTRIBUTED
+SELECT create_distributed_hypertable('test_setting', 'time', replication_factor => 2);
```

Comment:
Since https://github.com/timescale/timescaledb/pull/3753 was merged before this patch I'm afraid we have to add a couple of more tests:
- Compression of CAGGs with variable-sized buckets
- Compression of CAGGs with variable-sized buckets on top of distributed HT

If it doesn't work it's fine just to make sure that the user will see an adequate error message and add corresponding support to the backlog.

Reply:
I added a basic test to the new test file. But could you elaborate why we need a special test case for caggs with variable sized buckets? Compression is applied on the underlying storage i.e. on the materialized hypertable. time_bucket_ng is just another function used in the definition of the view. All the code in this PR is syntactic sugar so that users don't have to enable compression directly on the materialized hypertable.

Reply:
@gayyappan thanks! If we have a feature (compression of CAGGs) that works with N other features (CAGGs with variable buckets, CAGGs with fixed buckets, CAGGs on top of distributed HT) we have to make sure that they work together as expected (on all platforms, with different compile flags, etc). Otherwise, the code coverage is simply inadequate, although maybe it looks OK in terms of LoC. This allows to discover bugs early (before the support team discovers them) and simplifies the maintenance of the project in the long term.

Reply:
The point is caggs with variable size buckets or fixed size buckets are internals specific to the time_bucket function implementation and should affect only bucket computation logic (or places where bucket computation logic is needed). It does not affect anything related to how the materialized data is stored. As I stated in my previous comment, I added a test for sanity checking.

Reply:
The point is, unless we have at least the simplest tests, it's not proven that named functionalities are working together. Also, we will not catch it in the future if they stop working together.